Alpine ARM64 build fails because of old grpc version

<!--
Please note that the Vagrant issue tracker is reserved for bug reports. For general usage questions, please use
HashiCorp Discuss, https://discuss.hashicorp.com/c/vagrant/.

When you submit a bug report, please provide the minimal configuration and required information necessary to reliably reproduce the issue. It
should include a basic Vagrantfile.

Thank you!
-->

### Debug output

#14 1041.2 In file included from third_party/re2/util/pcre.cc:21:
#14 1041.2 third_party/re2/util/pcre.h:503:11: error: 'int32_t' does not name a type
#14 1041.2 503 |   mutable int32_t     hit_limit_;      // Hit limit during execution

####

1041.2 An error occurred while installing grpc (1.56.2), and Bundler cannot continue.
1041.2 
1041.2 In Gemfile:
1041.2   vagrant was resolved to 2.3.8.dev, which depends on
1041.2     grpc


<!--
Provide a link to a GitHub Gist containing the complete debug output, https://www.vagrantup.com/docs/other/debugging.html. 

The debug output should
be very long. Do NOT paste the debug output in the issue.
-->

### Expected behavior

Build should succeed in container for alpine:3.19.1.

<!--
What should have happened?
-->

### Actual behavior

<!--
What actually happened?
-->

Build failed.

### Reproduction information 

#### Vagrant version

2.4.1

<!--
Use `vagrant -v` to collect the version information. If you are not running the latest version
of Vagrant, please upgrade before submitting an issue.
-->

#### Host operating system

GNU/Linux

<!--
Your local system.
-->

#### Guest operating system

<!--
The operating system of the virtual machine.
-->

#### Steps to reproduce

1. Clone repository in alpine:3.19.1 based Dockerfile with required dependencies.
2. Clone Vagrant repo and execute bundle install.

I have traced the failure to the following diff where grpc has been pinned to 1.56: https://github.com/hashicorp/vagrant/compare/2.3.8.dev+000094-a7135c00...2.3.8.dev+000098-91191982#diff-93a1ba7172ffd60d0dda0a39a5c3538c0ca7b1b30efe1996c9dd38e2c78253d9R23

Before that change the builds were using grpc > 1.60 and there was no issue building in Alpine ARM64.

The core cause comes from google/re2 which has been fixed in the more recent releases, e.g.: https://github.com/google/re2/blob/main/util/pcre.h

Here is the change that was made: https://groups.google.com/g/re2-dev/c/HrOT3p8dNPw

Would it be possible to update grpc in main/nightly/next-release?

--- a/vagrant.gemspec
+++ b/vagrant.gemspec
@@ -20,7 +20,7 @@ Gem::Specification.new do |s|
   s.add_dependency "ed25519", "~> 1.3.0"
   s.add_dependency "erubi"
   s.add_dependency 'googleapis-common-protos-types', '~> 1.3'
-  s.add_dependency "grpc", "~> 1.56.0"
+  s.add_dependency "grpc", "~> 1.62.0"
   s.add_dependency "hashicorp-checkpoint", "~> 0.1.5"
   s.add_dependency "i18n", "~> 1.12"
   s.add_dependency "listen", "~> 3.7"

